src/cogent3/core/new_sequence.py

```diff
@@ -327,84 +329,80 @@ def strip_degenerate(self):
         """Removes degenerate bases by stripping them out of the sequence."""
         return self.__class__(
             moltype=self.moltype,
-            seq=self.moltype.strip_degenerate(self),
+            seq=self.moltype.strip_degenerate(str(self)),
```

Comment:
comment about refactor, perhaps `numpy.array(self)` more efficient?

Reply:
the underlying methods uses `bytes` so i've changed it to that